src/node_url.cc

```diff
@@ -93,6 +93,30 @@ void Parse(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(true);
 }
 
+void CanParse(const FunctionCallbackInfo<Value>& args) {
+  CHECK_GE(args.Length(), 2);
+  CHECK(args[0]->IsString());  // input
+  // args[1] // base url
+
+  Environment* env = Environment::GetCurrent(args);
+  HandleScope handle_scope(env->isolate());
+  Context::Scope context_scope(env->context());
+
+  Utf8Value input(env->isolate(), args[0]);
+  ada::result base;
+  ada::url* base_pointer = nullptr;
+  if (args[1]->IsString()) {
+    base = ada::parse(Utf8Value(env->isolate(), args[1]).ToString());
```

Comment:
The Utf8Value's lifetime only extends to the end of line. I can't quite tell what lifetime ada::parse() expects for its argument (only that it takes a string_view) so I'm not sure if this is correct.

Reply:
The lifetime expectency is correct on this line. After parsing, aka this line, parser never access the string_view